src/textual/widgets/_datatable.py

```diff
@@ -0,0 +1,214 @@
+from __future__ import annotations
+
+from dataclasses import dataclass
+from typing import ClassVar, Generic, TypeVar, cast
+
+from rich.padding import Padding
+from rich.segment import Segment
+from rich.style import Style
+from rich.text import Text, TextType
+
+from .._lru_cache import LRUCache
+from .._segment_tools import line_crop
+from .._types import Lines
+from ..geometry import Size
+from ..reactive import Reactive
+from ..scroll_view import ScrollView
+from ..widget import Widget
+
+CellType = TypeVar("CellType")
+
+
+@dataclass
+class Column:
+    label: Text
+    width: int
+    visible: bool = False
+    index: int = 0
+
+
+@dataclass
+class Cell:
+    value: object
+
+
+class Header(Widget):
+    pass
+
+
+class DataTable(ScrollView, Generic[CellType]):
+
+    CSS = """
+    DataTable {
+        background: $surface;
+        color: $text-surface;       
+    }
+    DataTable > .datatable--header {        
+        text-style: bold;
+        background: $primary;
+        color: $text-primary;
+    }
+    DataTable > .datatable--fixed {
+        text-style: bold;
+        background: $primary-darken-2;
+        color: $text-primary-darken-2;
+    }
+
+    DataTable > .datatable--odd-row {
+        
+    }
+
+    DataTable > .datatable--even-row {
+        background: $primary 10%;
+    }
+
+    DataTable > .datatable--highlight {
+        background: $secondary;
+        color: $text-secondary;
+    }
```

Comment:
This `highlight` style looks very interesting. I don't see it being used right now but played around with it by storing a `current_row` on the `DataTable` instance which will use this instead of the even/odd style.
The `TableApp` can then update the `current_row` value when the user presses up/down (j/k, etc.). Was this potentially an intended use case here?
If so, it might make more sense to track the `current_row` on the App, rather than the internal table instance. This would require the option to change the style of a given row individually. Is something like that planned/in-scope?